File: object-oriented-programming/virtual-world-python/plants/grass.py
from plants.plant import Plant
from species import Species
import random

class Grass(Plant) :

    def __init__(self, world, x=None, y=None) :
        if x == None and y == None :
            super().__init__(Species.GRASS, 0, 0, 0, world, 1)
            width = world.width
            height = world.height

            while(True):
                new_x = random.randint(0, width - 1)
                new_y = random.randint(0, height - 1)
                if self._world.is_free(new_x, new_y) :
                    break
                else :
                    continue

            self._world.set_organism(None, self.x, self.y)
            self._x = new_x
            self._y = new_y
            self._world.set_organism(self, self.x, self.y)
            return
        
        super().__init__(Species.GRASS, x, y, 0, world, 1)

    def action(self) :
        return super().action()

    def collision(self, other):
        return super().collision(other)

    def draw_color(self):
        return self.species.color
    
    def sow(self, x, y):
        return Grass(self._world, x, y)

File: object-oriented-programming/virtual-world-java/src/Game/Model/Plants/Grass.java
package Game.Model.Plants;

import Game.Model.Species;
import Game.Model.World;

import java.awt.*;
import java.util.Random;

public class Grass extends Plant {
    Random random = new Random();

    public Grass(int x, int y, World world){
        super(Species.GRASS, x, y, 0, world, 1);
    }

    public Grass(World world){
        super(Species.GRASS, 0, 0, 0, world, 1);
        int width = world.getWidth();
        int height = world.getHeight();

        int newX, newY;

        do {
            newX = random.nextInt(width);
            newY = random.nextInt(height);
        } while(!world.isFree(newY, newX));

        world.setOrganism(null, y, x);
        x = newX;
        y = newY;
        world.setOrganism(this, y, x);
    }

    public final Color draw(){
        return species.displayColor();
    }

    public Grass sow(int y, int x){
        return new Grass(x, y, world);
    }
}
